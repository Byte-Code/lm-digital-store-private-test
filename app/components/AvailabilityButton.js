import React, { Component, PropTypes } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import Dialog from 'material-ui/Dialog';
import styled from 'styled-components';

import AvailabilityDialog from './AvailabilityDialog';

const Button = styled.div`
  width: '100%';
  text-transform: uppercase;
  background: ${props => props.bgColor};
  height: 60px;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  color: #fff;
  box-shadow:  0 0 8px 0 rgba(51, 51, 51, 0.1);
  cursor: pointer;
`;

export default class AvailabilityButton extends Component {
  static propTypes = {
    productName: PropTypes.string.isRequired,
    productCode: PropTypes.string.isRequired,
    nearbyStoreStock: ImmutablePropTypes.list.isRequired,
    nearbyStores: ImmutablePropTypes.list.isRequired
  }
  constructor(props) {
    super(props);
    this.state = {
      dialogOpen: false
    };
  }

  handleOpen = () => {
    this.setState({ dialogOpen: true });
  }

  handleClose = () => {
    this.setState({ dialogOpen: false });
  }

  render() {
    const {
      nearbyStores,
      nearbyStoreStock,
      productName,
      productCode
    } = this.props;

    return (
      <div>
        <Button
          bgColor="#67cb33"
          onClick={this.handleOpen}
        >
          verifica disponibilit√† in negozi vicini
        </Button>
        <Dialog
          modal={false}
          onRequestClose={this.handleClose}
          open={this.state.dialogOpen}
          contentStyle={{ width: 1000 }}
          bodyStyle={{ padding: '80px 75px', background: '#333333' }}
        >
          <AvailabilityDialog
            nearbyStores={nearbyStores}
            nearbyStoreStock={nearbyStoreStock}
            productName={productName}
            productCode={productCode}
          />
        </Dialog>
      </div>
    );
  }
}
