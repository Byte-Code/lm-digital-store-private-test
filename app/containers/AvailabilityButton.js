import { connect } from 'react-redux';

import AvailabilityButton from '../components/AvailabilityButton';
import { getNearbyStores } from '../reducers/selectors';

const mapStateToProps = (state) => ({
  nearbyStores: getNearbyStores(state)
});

export default connect(mapStateToProps, {})(AvailabilityButton);
