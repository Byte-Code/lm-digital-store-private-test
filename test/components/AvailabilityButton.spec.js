import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import AvailabilityButton, { Button } from '../../app/components/AvailabilityButton';

describe('AvailabilityButton', () => {
  const productName = 'abc';
  const productCode = '123';

  it('should render properly with a closed dialog', () => {
    const nearbyStoreStock = fromJS([0, 1, 2]);
    const result = shallow(
      <AvailabilityButton
        productCode={productCode}
        productName={productName}
        nearbyStoreStock={nearbyStoreStock}
      />
    );
    expect(result).toMatchSnapshot();
    expect(result.state('dialogOpen')).toBe(false);
  });

  it('should display a dialog when the button is clicked', () => {
    const nearbyStoreStock = fromJS([0, 1, 2]);
    const result = shallow(
      <AvailabilityButton
        productCode={productCode}
        productName={productName}
        nearbyStoreStock={nearbyStoreStock}
      />
    );
    result.find(Button).simulate('click');
    expect(result.state('dialogOpen')).toBe(true);
    expect(result).toMatchSnapshot();
  });
});
