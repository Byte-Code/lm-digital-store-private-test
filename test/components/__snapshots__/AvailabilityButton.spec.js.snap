// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvailabilityButton should close the dialog once the closeButton is clicked 1`] = `
<div>
  <styled.div
    bgColor="#67cb33"
    onClick={[Function]}
  >
    verifica disponibilità in negozi vicini
  </styled.div>
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "#333333",
        "padding": "75px 70px",
      }
    }
    contentStyle={
      Object {
        "maxWidth": "none",
        "width": 1000,
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
  >
    <CloseButton
      backgroundColor="#333333"
      fill="#858585"
      handleClick={[Function]}
      right={0}
      top={0}
    />
    <AvailabilityDialog
      nearbyStoreStock={
        Array [
          0,
          1,
          2,
        ]
      }
      productCode="123"
      productName="abc"
    />
  </Dialog>
</div>
`;

exports[`AvailabilityButton should display a dialog when the button is clicked 1`] = `
<div>
  <styled.div
    bgColor="#67cb33"
    onClick={[Function]}
  >
    verifica disponibilità in negozi vicini
  </styled.div>
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "#333333",
        "padding": "75px 70px",
      }
    }
    contentStyle={
      Object {
        "maxWidth": "none",
        "width": 1000,
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={true}
    repositionOnUpdate={true}
  >
    <CloseButton
      backgroundColor="#333333"
      fill="#858585"
      handleClick={[Function]}
      right={0}
      top={0}
    />
    <AvailabilityDialog
      nearbyStoreStock={
        Array [
          0,
          1,
          2,
        ]
      }
      productCode="123"
      productName="abc"
    />
  </Dialog>
</div>
`;

exports[`AvailabilityButton should render null when nearbyStoreStock is empty 1`] = `null`;

exports[`AvailabilityButton should render properly with a closed dialog when nearbyStoreStock is not Empty 1`] = `
<div>
  <styled.div
    bgColor="#67cb33"
    onClick={[Function]}
  >
    verifica disponibilità in negozi vicini
  </styled.div>
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "#333333",
        "padding": "75px 70px",
      }
    }
    contentStyle={
      Object {
        "maxWidth": "none",
        "width": 1000,
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
  >
    <CloseButton
      backgroundColor="#333333"
      fill="#858585"
      handleClick={[Function]}
      right={0}
      top={0}
    />
    <AvailabilityDialog
      nearbyStoreStock={
        Array [
          0,
          1,
          2,
        ]
      }
      productCode="123"
      productName="abc"
    />
  </Dialog>
</div>
`;
